astropy/units/structured.py

```diff
@@ -3,6 +3,8 @@
 This module defines structured units and quantities.
 """
 
+from __future__ import annotations
```

Comment:
Is there an associated version of python when this will no longer be needed?

Reply:
Yes. The nice thing is that ``pyupgrade`` in ``pre-commit`` should fix this when we bump the minimum version and run ``pre-commit`` on the repository.
Maybe that should be a step in the release process...

Reply:
Is the line here just to avoid importing `typing.Union`?

Reply:
Currently, yes.

Reply:
[PEP 563](https://peps.python.org/pep-0563/) has been [postponed indefinitely](https://docs.python.org/3/library/__future__.html#id2), so I wouldn't expect `pyupgrade` to remove the line in question any time soon. However, writing union types as `X | Y` instead of `Union[X, Y]` proposed by [PEP 604](https://peps.python.org/pep-0604/) was adopted in Python 3.10, so if the import is only needed for that then it could be removed when support for Python 3.9 is dropped.

Reply:
Thanks for clearing up the timeline!

Reply:
Maybe mark with a comment `# For python < 3.10.`?